src/openvic-simulation/dataloader/NodeTools.cpp

```diff
@@ -142,8 +142,8 @@ node_callback_t NodeTools::expect_colour(callback_t<colour_t> callback) {
 	};
 }
 
-node_callback_t NodeTools::expect_date(callback_t<Date> callback) {
-	return expect_identifier(
+node_callback_t NodeTools::expect_date(callback_t<Date> callback, string_conversion_callback_t parse_callback) {
```

Comment:
Solved this problem differently with OOBs, just changed `expect_date` to use `expect_identifier_or_string` which feels more concise to use. This method works too but we need to decide on one to use.

Reply:
Could change the default to be `expect identifier_or_string` and then introduce the parse_callback for any additional behavior instead, that's probably the best solution.

Reply:
Sounds good to me

Reply:
The approach I've used for things like this is to return ‘std::string_view‘ callback, so you can then choose to do ‘expect_identifier(expect_date(...))‘ or ‘expect_string(expect_date(...))‘, or any other function which accepts a string_view callback. This also removes the need for a Node as the root source of the date

Reply:
Thinking about it some more, every expect function which is basically just # a specialisation of ‘expect_identifier‘ should return a string_view callback, eg expect_fixed_point.